addons/l10n_pk_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,420 @@
+                if line.discount < 0:
+                    error_message.append(_("Negative discount is not allowed, set in line %s", line.name))
+                hs_code = self._l10n_pk_edi_extract_digits(line.product_id.hs_code)
+                if not hs_code:
+                    error_message.append(_("HS code is not set in product line %s", line.name))
+                elif not re.match("^[a-zA-Z0-9]{8}$", hs_code):
+                    error_message.append(_(
+                        "Invalid HS Code (%(hs_code)s) in product line %(product_line)s",
+                            hs_code=hs_code,
+                            product_line=line.product_id.name or line.name
+                        )
+                    )
+        return error_message
+
+    def _needs_web_services(self):
+        self.ensure_one()
+        return self.code == "pk_einvoice_1_3" or super()._needs_web_services()
+
+    def _l10n_pk_edi_xml_invoice_content(self, invoice):
+        return json.dumps(self._l10n_pk_edi_generate_invoice_json(invoice)).encode()
+
+    def _l10n_pk_validate_partner(self, partner, is_company=False):
+        self.ensure_one()
+        message = []
+        if not re.match("^.{3,100}$", partner.street or ""):
+            message.append(_("- Street required min 3 and max 100 characters"))
+        if partner.street2 and not re.match("^.{3,100}$", partner.street2):
+            message.append(_("- Street2 should be min 3 and max 100 characters"))
+        if not re.match("^.{3,100}$", partner.city or ""):
+            message.append(_("- City required min 3 and max 100 characters"))
+        if partner.country_id.code == "PK" and not re.match("^.{3,50}$", partner.state_id.name or ""):
+            message.append(_("- State required min 3 and max 50 characters"))
+        if partner.country_id.code == "PK" and not re.match("^[0-9]{6}$", partner.zip or ""):
+            message.append(_("- Zip code required 6 digits"))
+        if partner.phone and not re.match("^[0-9]{10,12}$",
+            self._l10n_pk_edi_extract_digits(partner.phone)
+        ):
+            message.append(_("- Mobile number should be minimum 10 or maximum 12 digits"))
+        if partner.email and (
+            not re.match(r"^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$", partner.email)
+            or not re.match("^.{6,100}$", partner.email)
+        ):
+            message.append(_("- Email address should be valid and not more then 100 characters"))
+        if message:
+            message.insert(0, "%s" % (partner.display_name))
+        return message
+
+    def format_timestamp(self, value):
+        return value.strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3] + 'Z'
+
+    @api.model
+    def _l10n_pk_round_value(self, amount, precision_digits=2):
+        """
+            This method is call for rounding.
+            If anything is wrong with rounding then we quick fix in method
+        """
+        return round(amount, precision_digits)
```

Comment:
Why not using round directly?

Reply:
I created the `_l10n_pk_round_value` method instead of using Python's `round` function directly to make the code more flexible and easier to maintain. By keeping the rounding logic in one place, we can quickly update the rules or precision in the future without changing multiple parts of the code. This makes it simpler to handle any changes that might come up. 🤔

Reply:
```suggestion
        return round(amount, precision_digits) or 0.0  # avoid -0.0
```
`_l10n_pk_round_value` would make sense with this rather than this it's seems useless to me